src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/ModuleTokenResolver.cs

```diff
@@ -60,6 +60,12 @@ public ModuleToken GetModuleTokenForType(EcmaType type, bool throwIfNotFound = t
                 return token;
             }
 
+            // If the token was not lazily mapped, search the input compilation set for a type reference token
+            if (_compilationModuleGroup.TryGetModuleTokenForExternalType(type, out token))
```

Comment:
Is this guaranteed to be called from a single thread only?

Reply:
The signature emission happens in a single thread at the end (otherwise we can have determinism bugs).

Reply:
Verified under debugger too.